### PR TITLE
fix: retrieve location correctly in geo location flow

### PR DESCRIPTION
**Pull request by khairul-alam-licon**
# Description
Fix issue when location was showing during custom permission is denied but device permission is allowed. It seems that `onGeolocationPermissionsShowPrompt` has an intention to check permission state for `android.permission.ACCESS_FINE_LOCATION`, so we should add a barricade of custom permission while invoking the permission which affects Geolocation API. 

Initially, I intended to show dialogs for both device and custom permission before executing `GeolocationPermissions.Callback?.invoke(origin, true, false)`, but it seems the checkup of JS SDK is already enough. So, in Android side, I have used the value of custom permission under `onGeolocationPermissionsShowPrompt` at runtime.

## Links
- MINI-2314

# Checklist
- [x] I have read the [contributing guidelines](../.github/CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](../.github/CONTRIBUTING.md#SDK-Development-Learning-Path) (if submitting a feature PR)
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/MiniAppWebChromeClient.kt [hunk 1]
```diff
@@ -38,7 +45,19 @@ internal class MiniAppWebChromeClient(
...
     ) {
-        callback?.invoke(origin, true, false)
+        val hasCustomPermission =
+            miniAppCustomPermissionCache.readPermissions(miniAppInfo.id)
+                .pairValues.find {
+                    it.first == MiniAppCustomPermissionType.LOCATION
+                }?.let {
+                    it.second == MiniAppCustomPermissionResult.ALLOWED
+                }
+        val hasDevicePermission = ActivityCompat.checkSelfPermission(
```
Comment (corycaywood) [inline miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/MiniAppWebChromeClient.kt:55]: Do we need to have this check for the device permission? I think this is already handled by the android system. Device permissions was already working correctly without this.

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/MiniAppWebChromeClient.kt [hunk 2]
```diff
@@ -38,7 +45,19 @@ internal class MiniAppWebChromeClient(
         origin: String?,
         callback: GeolocationPermissions.Callback?
     ) {
-        callback?.invoke(origin, true, false)
+        val hasCustomPermission =
+            miniAppCustomPermissionCache.readPermissions(miniAppInfo.id)
```
Comment (corycaywood) [inline miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/MiniAppWebChromeClient.kt:49]: It seems like we're duplicating code in several places for checking if a permission is granted (for example in `CustomPermissionBridgeDispatcher`, in `UserInfoBridgeDispatcher`, and right here).

Maybe we should add something like `isGranted(permission: MiniAppCustomPermissionType)` to `MiniAppCustomPermissionCache`? I guess this could be a refactoring improvement in another PR.

**Review by corycaywood — COMMENTED** [review]
Looking good 👍  See my comment below about the native permission.

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/MiniAppWebChromeClient.kt (hunk 1 above)
Comment (khairul-alam-licon) [reply miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/MiniAppWebChromeClient.kt:55]: Thanks for the concern, we can skip this also. Initially, I thought we should check this but it's not obvious in this situation, so I'll update this part.

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/MiniAppWebChromeClient.kt (hunk 2 above)
Comment (khairul-alam-licon) [reply miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/MiniAppWebChromeClient.kt:49]: Right, I have added something like `hasPermission(...)` to `MiniAppCustomPermissionCache` for this cleanup.
It can be seen in https://github.com/rakutentech/android-miniapp/pull/222

**Review by corycaywood — APPROVED** [review]
Looks good! 👍

**Review by minh-rakuten — APPROVED** [review]
👍 👍
